src/arch/threadbase.h

```diff
@@ -89,7 +89,7 @@ namespace forte {
 
         /*! \brief set the alive flag for this flag
          */
-        void setAlive(bool paVal) {
+        virtual void setAlive(bool paVal) {
```

Comment:
I'm not sure if I'm happy with that change. This method is on purpose not virtual.

Reply:
I had to check a second time, because I thought we have a generic solution. But unfortunately when we did the last redesign of our thread infra we decided against a generic solution. AFAIR the reason was that the different use cases are to specific each time. On solution you can see in our CEventChainExecutionThread class. But after your feedback I'm not to happy about it anymore.

One thing that came to my mind is to introduce a new private virtual method `onEnd()` which threads could override to perform closing specific things. This method could be called from the end() method.

I think private is important because only CThreadBase should be allowed to call this method.

Reply:
@azoitl I respect your considerations. For the time being, may we keep my solution and revert/fix it once the greater refactoring takes place? Thank you for your effort.

Reply:
@dok-net as this is one of the most curcial components in 4diac FORTE I would not like to have this intermediate fix in it. However I can offer to do the proposed refactoring upfront your PR.

Reply:
@azoitl Yes, please, by all means, go ahead with that.

Reply:
@dok-net done in PR #41. BTW thx for finding this problem. With the new api this is much cleaner and made CEventChainExecutionThread more correct.

Reply:
The modular IO core doesn't seem to correctly support asynchronous input... It keeps emitting two IND events at the ID where only one would be justified. That is for cyclical updates triggered from `runLoop()`. I am out of my depth how to fix this without modifying the core implementation, which is quite complicated at that.
I am referring to the IOHandleADC compiled for async ADC... the runLoop() causes a check of equal, which return false, in order to trigger a get. Once the async measurement started by get finishes, work_callback triggers another get. Which functions exactly I am using, is all visible in this PR. Unfortunately, this sequence of execution results in ID's emitting of TWO IND events, instead of one. Yes, it reads twice. But once is just perfunctory to even start the measurement. Is the modular IO not designed to support async input, that has to be actively triggered? ADC must be started, as opposed to GPIO interrupts...

Reply:
I think the problem of your solution is that you use the default `checkForInputChanges();`  will update and fire IND events for all registered FBs. This is maybe not what you want. So calling this default method from you runLoop is maybe not the right thing. The other option is that you store your ADC handles in a seperate list and manage them seperatly.

Reply:
I fixed it. Now REQ for ADC results in both CNF and IND, whereas cyclical sampling from the device configuration FB result in IND, only. This allows the app to connect to only IND in most cases, alleviating the need to deal with multiple events for the same measurement.

Reply:
Interesting approach

Reply:
Did the same for GPIO now - IND for any change of value, be it via IRQ or FB REQ event.

Reply:
@azoitl This requested change was resolved, please re-review and close it accordingly, if you are OK with it.